ui-tests/src/test/java/io/syndesis/qe/pages/apps/todo/Todo.java

```diff
@@ -1,13 +1,53 @@
 package io.syndesis.qe.pages.apps.todo;
 
-import static com.codeborne.selenide.Condition.visible;
+import static com.codeborne.selenide.Selenide.$;
 import static com.codeborne.selenide.Selenide.$$;
 
+import com.codeborne.selenide.Condition;
+import com.codeborne.selenide.ElementsCollection;
+import com.codeborne.selenide.Selenide;
+import com.codeborne.selenide.SelenideElement;
+import org.assertj.core.api.Assertions;
 import org.openqa.selenium.By;
 
 public class Todo {
 
+    public void refresh(){
+        Selenide.refresh();
+    }
+
     public int getListItemsCount() {
-        return $$(By.xpath("//li[@class='list-group-item']//button[@value='check']")).size();
+        return getListItems().size();
+    }
+
+    public ElementsCollection getListItems(){
+        return $$(By.xpath("//li[@class='list-group-item']//input[@name='task' and @value!='']"));
+    }
+
+    public void setJmsForm(String text){
+        showFormIsNecessary();
+        SelenideElement textArea = $(By.name("message")).shouldBe(Condition.visible);
+        textArea.clear();
+        textArea.sendKeys(text);
+    }
+
+    public void sentJmsMessage(){
+        showFormIsNecessary();
+        SelenideElement button = $(By.name("amq")).shouldBe(Condition.visible);
+        button.click();
+        Assertions.assertThat($(By.xpath("//*[@id=\"jmsToggle\"]/div/ul/li/div/span")).text()).contains("Message sent successfully");
+    }
+
+    public String getMessageFromTodo(int index){
+        ElementsCollection lists = getListItems();
+        SelenideElement item = lists.get(index);
+        return item.getValue();
+    }
+
+    public void showFormIsNecessary(){
```

Comment:
so ... the method does show "form is necessary"? :-) the naming is odd

Reply:
typo there. There should be showFormIfNecessary but of course, it is necessary when it was called :D .. so I rewrite it just showJmsForm()